src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -115,6 +130,247 @@ public static Task<List<string>> ReadRequestAndSendResponseAsync(Socket server,
             return lines;
         }
 
+        public static async Task<List<string>> ValidateAuthenticationAsync(Socket s, StreamReader reader, StreamWriter writer, string response, Options options)
+        {
+            // Send unauthorized response from server.
+            await ReadWriteAcceptedAsync(s, reader, writer, response);
+
+            // Read the request method.
+            string line = await reader.ReadLineAsync().ConfigureAwait(false);
```

Comment:
Just FYI, having it won't hurt anything, but we generally don't use ConfigureAwait in test code.

Reply:
Why not?  Is this just for the ReadLineAsync() method or generally?  Wouldn't we have the same deadlock problems in test code as in product code?

Reply:
> Is this just for the ReadLineAsync() method or generally?

Generally

> Wouldn't we have the same deadlock problems in test code as in product code?

We shouldn't be writing tests that synchronously block on such helpers; if we do, that's a test bug.  We use ConfigureAwait(false) in product code for perf but also because we can't control whether someone does synchronously wait on it.

> Why not?

Three reasons:
1. xunit has a synchronization context in place for a reason, and we generally want to respect that reason, which is to limit concurrency.
2. ConfigureAwait(false) is partially for eeking out additional perf, which we don't care about in tests, and it adds clutter that's only worthwhile if it brings value.
3. We shouldn't go out of our way in test code to enable other test code to block artificially and introduce deadlock potential.

Reply:
This is really good information about testing and ConfigureAwait().  I don't think the rest of the CoreFx community knows this.  It would be good to distribute this information more broadly.

cc: @caesar1995 @rmkerr